src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -253,6 +267,22 @@ public void testRequiredTokenIsEmptyIntArray() throws Exception {
         }
     }
 
+    @Test
+    public void testBehaviourWithZeroChecks() throws Exception {
+        final String errMsg = "No checks -> No parsing";
+        final TreeWalker treeWalkerSpy = spy(new TreeWalker());
+        final Class<?> classAstState =
+                Class.forName("com.puppycrawl.tools.checkstyle.TreeWalker$AstState");
+        mockStatic(TreeWalker.class);
+        when(TreeWalker.parse(any(FileContents.class)))
+                .thenThrow(new IllegalStateException(errMsg));
+        doNothing().when(treeWalkerSpy, "walk",
+                any(DetailAST.class), any(FileContents.class), any(classAstState));
+        treeWalkerSpy.processFiltered(temporaryFolder.newFile("file.java"), new ArrayList<>());
+        verifyPrivate(treeWalkerSpy, times(0)).invoke("walk",
```

Comment:
@ps-sp 
I am aware it is the same functionally, but it improves readability, like `assertEquals(a, b)` vs `assertTrue(a.equals(b))`